blog-posts/getting-started-gatsby-next-remix.md

```diff
@@ -0,0 +1,306 @@
+
+
+## Setting up a project
+
+#### Gatsby
+
+To start a Gatsby project you have to run `npx gatsby new`. The setup process offers many individualization possibilities: Besides the project name and path, a CMS, a styling system (e.g. styled-components, emotion, ...), and additional features with other plugins (e.g. mdx support) can be added optionally. Gatsby has built-in out-of-the-box TypeScript support, so there is no distinction between TypeScript and JavaScript in the project setup.
+
+#### Next.js
+
+The command `npx create-next-app@latest` can additionally be extended with the flag `--ts` to add direct TypeScript support to a new Next.js project. The following setup process does not provide any customization options apart from the project name.
+
+#### Remix
+
+A Remix project can be set up with the command `npx create-remix@latest`. Then you can specify the project name, choose between different host options (e.g. Netlify or Vercel) and between TypeScript and JavaScript.
+
+## Static Site Generation
+
+With static site generation (SSG), the HTML is generated once at built time and reused for each request. Of the three frameworks in this article, only Remix does not support SSG.
+
+
+> ⚡ The options shown below for loading data only works in page/template components (they are bound to a certain url path or directory). To load data in components that do not have their own route, you must use other methods
+
+
+#### **Gatsby**
+
+A static generated page in Gatsby consists of two parts: A GraphQL query and the actual React component (here `BlogPage`), which must be exported as default. In the GraphQL query, the required data is fetched. In this example, the Gatsby plugin [gatsby-transformer-remark](https://www.gatsbyjs.com/plugins/gatsby-transformer-remark/) is used to get the content of the corresponding markdown file.
+In the actual component, the results of the query can then be accessed via `data`.
+
+In our Gatsby blog project, we created a file named `{markdownRemark.frontmatter__path}.tsx` to create a route for each blogpost. The important thing here is that the filename is the path of the query in curly brackets. This way the data from the markdown files determines the routes of the project.
+
+```tsx
+const BlogPost = ({data}) => {
+    return (
+        <div dangerouslySetInnerHTML={{__html: data.markdownRemark.html}}/>
+    )
+}
+
+export const query = graphql`
+    query ($id: String) {
+        markdownRemark(id: {eq: $id}) {
+            frontmatter {
+                title
+                path
+            }
+            html
+        }
+    }
+`
+
+export default BlogPost
+```
+
+
+> 💡 Alternatively, pages can also be created in `gatsby-node.js`. For this, the `createPages` API provided by Gatsby must be used. For more information on this, as well as an easy-to-understand example, see the [Gatsby documentation](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#creating-pages-in-gatsby-nodejs).
```

Comment:
This is explicit about manually create pages with `createPages` vs implicit (gatsby page plugin). I think you should write a paragraph about this differentiation because it's pretty confusing here. Maybe it's related to my earlier confusion about the template you display itself ? (see other comment)

Reply:
I've added an example with the create-pages API too 👍